src/Symfony/Component/Validator/Tests/Constraints/FileValidatorTestCase.php

```diff
@@ -658,5 +658,47 @@ public function testUploadedFileExtensions()
         $this->assertNoViolation();
     }
 
+    /**
+     * @dataProvider filenameMaxLengthIsTooLongProvider
+     */
+    public function testFilenameMaxLengthIsTooLong(File $constraintFile, string $messageViolation)
+    {
+        file_put_contents($this->path, '1');
+
+        $file = new UploadedFile($this->path, 'myFileWithATooLongOriginalFileName', null, null, true);
+        $this->validator->validate($file, $constraintFile);
+
+        $this->buildViolation($messageViolation)
+            ->setParameters([
+                '{{ filename_max_length }}' => $constraintFile->filenameMaxLength,
+            ])
+            ->setCode(File::FILENAME_TOO_LONG)
+            ->assertRaised();
+    }
+
+    private function filenameMaxLengthIsTooLongProvider(): \Generator
```

Comment:
providers should be static + this should be named provideFilenameMaxLengthIsTooLong

Reply:
Changed ! Thanks too !